invenio_workflows_tugraz/theses/theses.py

```diff
@@ -184,14 +183,8 @@ def update_func(
     identity: Identity,
 ) -> None:
     """Update the record by metadata from alma."""
-    try:
-        data = records_service.read(id_=marc_id, identity=identity).data
-    except PIDDoesNotExistError:
-        try:
-            data = records_service.read_draft(id_=marc_id, identity=identity).data
-        except PIDDoesNotExistError as esc:
-            msg = f"pid: {marc_id} not found"
-            raise RuntimeError(msg) from esc
+    record_list = records_service.search_versions(id_=marc_id, identity=identity)
+    data = list(record_list.hits)[0]
```

Comment:
if a wrong `marc_id` is used, does exists the hits in record_list or could it be empty ?

Reply:
if a wrong marc_id is used, then the search_versions should raise an NoResultFound exception which leads to not executed hits statement

Reply:
that is true, but if hits is empty and list(record_list.hits)[0]) would raise an exception, then it is also ok, because the further could would not be executed and no harm on the database would occur. the question is how to handle the exception on the update_func call side. but that is another question